src/components/ParsedBotMessageBody.tsx

```diff
@@ -90,3 +123,45 @@ export default function ParsedBotMessageBody(props: Props) {
   }
   return <Text style={{ borderRadius: 16 }}>{message.message}</Text>;
 }
+
+const urlRegex =
+  /(?:https?:\/\/|www\.)?[-a-zA-Z0-9@:%._+~#=]{1,256}\.(xn--)?[a-z]{2,20}\b([-a-zA-Z0-9@:%_+[\],.~#?&/=]*[-a-zA-Z0-9@:%_+~#?&/=])*/g;
+const markdownUrlRegex = /\[(.*?)\]\((.*?)\)/g;
+
+interface RegexTextPattern {
+  regex: RegExp;
+  replacer(params: {
+    match: string;
+    groups: string[];
+    index: number;
+  }): string | ReactNode;
+}
+const RegexText = ({
+  children,
+  patterns,
+}: {
+  children: string;
+  patterns: RegexTextPattern[];
+}) => {
+  if (patterns.length === 0 || typeof children !== 'string') {
+    return <>{children}</>;
+  }
+
+  const convertedNodes: Array<string | ReactNode> = [children];
+  patterns.forEach(({ regex, replacer }) => {
+    const node = convertedNodes.concat();
+    let offset = 0;
+    node.forEach((text, index) => {
+      if (typeof text === 'string' && text) {
+        const children = replaceWithRegex(text, regex, replacer);
+
+        if (children.length > 1) {
+          convertedNodes.splice(index + offset, 1, ...children);
```

Comment:
children is a string right? why we need spread operator here?

Reply:
children is a `Array<string | ReactNode>` type